storage/localstore/gc.go

```diff
@@ -50,6 +50,12 @@ func (db *DB) collectGarbageWorker() {
 	for {
 		select {
 		case <-db.collectGarbageTrigger:
+			// run through the recently pinned chunks and
+			// remove them from the gcIndex before calling GC
+			err := db.removeChunksInExcludeIndexFromGC()
```

Comment:
There is still a possibility that a chunk is pinned and added to the exclude index in between removeChunksInExcludeIndexFromGC and collectGarbage functions are called in this block, resulting in chunk removal even if it is pinned. I think that the same batchMu lock should be used for both functions. They both are locking the same mutex, but also releasing it in between.

Reply:
True. There is a slight chance..Fixed it now with single lock.

Reply:
Thanks!